src/commctl/kubeconfig.py

```diff
@@ -0,0 +1,146 @@
+    """
+    Base error for all KubeConfig errors.
+    """
+    pass
+
+
+class KubeConfigInvalidFileError(KubeConfigError):
+    """
+    Used then the kubeconfig file is invalid.
+    """
+    pass
+
+
+class KubeConfigNoContextError(KubeConfigError):
+    """
+    Used when there is no context set.
+    """
+    pass
+
+
+class KubeConfigUserTypeError(KubeConfigError):
+    """
+    Used when a user type can not be identified.
+    """
+    pass
+
+
+class KubeConfigNoUserError(KubeConfigError):
+    """
+    Used when a user can not be found.
+    """
+    pass
+
+
+class KubeUser:
+    """
+    Abstraction for a Kubernetes user.
+
+    Each instance will have the following attributes:
+
+    - name: The name of the user.
+    - type: The type of the credentials: token, password, client-certificate
+    - auth_headers: Additional authentication headers which should be used.
+    """
+
+    def __init__(self, spec):
+        """
+        Initializes a new KubeUser instance.
+
+        :param spec: A kubeconfig user spec.
+        :type spec: dict
+        :raises: commctl.kubeconfig.KubeConfigUserTypeError
+        """
+        self.name = spec['name']
+
+        if spec['user'].get('token'):
+            self.token = spec['user']['token']
+            self.auth_headers = [('Authorization', 'Bearer ' + self.token)]
+            self.type = 'token'
+        elif spec['user'].get('username') and spec['user'].get('username'):
+            import base64
+            self.username = spec['user']['username']
+            self.password = spec['user']['password']
+            self.auth_headers = [(
+                'Authorization', base64.encodestring(
+                    '{}{}'.format(self.username, self.password)))]
+            self.type = 'password'
+        elif (spec['user'].get('client-certificate') and
+                spec['user'].get('client-key')):  # pragma: no cover
+            self.client_certificate = os.path.realpath(
+                spec['user']['client-certificate'])
+            self.client_key = os.path.realpath(spec['user']['client-key'])
+            # TODO
+            self.auth_headers = ()
+            self.type = 'client-certificate'
+        else:
+            raise KubeConfigUserTypeError(
+                'Unknown user type: {0}'.format(spec))
+
+
+class KubeConfig:
+    """
+    Abstraction for a kubeconfig file.
+    """
+
+    def __init__(self, file_path):
+        """
+        Initializes an instance of KubeConfig.
+
+        :param file_path: Path to a kubeconfig file.
+        :type file_path: str
+        """
+        self.file_path = os.path.realpath(file_path)
+        with open(self.file_path, 'r') as f:
+            try:
+                self._kubeconfig = yaml.safe_load(f)
+            except yaml.error.YAMLError as error:
+                raise KubeConfigInvalidFileError(error)
+
+    @property
+    def current_context(self):
+        """
+        Returns the current context.
```

Comment:
I see `:raises:` used below.  Should be used here too I guess for `KubeConfigNoContextError`?